lib/url.rs

```diff
@@ -0,0 +1,104 @@
+use url;
+
+#[derive(Eq, Ord, Clone, Hash, PartialEq, PartialOrd, Serialize, Deserialize, Debug)]
+pub struct url_Url {
+    url: url::Url,
+}
+
+impl Default for url_Url {
+    fn default() -> Self {
+        url_Url {
+            url: url::Url::parse("http://127.0.0.1/").unwrap(),
+        }
+    }
+}
+
+impl FromRecord for url_Url {
+    fn from_record(val: &record::Record) -> result::Result<Self, String> {
+        match (val) {
+            record::Record::String(s) => match (url::Url::parse(&s)) {
+                Ok(url) => Ok(url_Url { url }),
+                Err(e) => Err(format!("{}", e)),
+            },
+            _ => Err(String::from("Unexpected type")),
+        }
+    }
+}
+
+impl IntoRecord for url_Url {
+    fn into_record(self) -> record::Record {
+        record::Record::String(self.url.as_str().to_string())
+    }
+}
+
+impl record::Mutator<url_Url> for record::Record {
+    fn mutate(&self, t: &mut url_Url) -> result::Result<(), String> {
+        *t = url_Url::from_record(self)?;
+        Ok(())
+    }
+}
+
+pub fn url_url_parse(s: &String) -> std_Result<url_Url, String> {
```

Comment:
I wonder whether we could generate some of this code automatically from the dl declarations.
Perhaps we could have an annotations for the ones where we can.